test/integration/httproute_webhook_test.go

```diff
@@ -139,13 +87,199 @@ func TestHTTPRouteValidationWebhook(t *testing.T) {
+		invalidRegexInPathTestCase(managedGateway, `HTTPRoute failed schema validation: schema violation (paths.3: invalid regex: '/foo[[[['`),
+		// No test case for invalid regex in header, because Kong Gateway doesn't return any error in such case (it works only for expressions router).
+	)
+	testHTTPRouteValidationWebhook(ctx, t, namespace, gatewayClient, testCases)
+}
+
+func TestHTTPRouteValidationWebhookExpressionsRouter(t *testing.T) {
+	skipTestForNonKindCluster(t)
+	skipTestForRouterFlavors(t, traditional, traditionalCompatible)
+
+	ctx := context.Background()
+	namespace, gatewayClient, managedGateway, unmanagedGateway := setUpEnvForTestingHTTPRouteValidationWebhook(ctx, t)
+	testCases := append(
+		commonHTTPRouteValidationTestCases(managedGateway, unmanagedGateway),
+		invalidRegexInPathTestCase(managedGateway, "regex parse error:\n    ^/foo[[[[\n            ^\nerror: unclosed character class)"),
+		testCaseHTTPRouteValidation{
+			Name: "a httproute with invalid regex for header does not pass validation",
+			Route: &gatewayv1beta1.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: uuid.NewString(),
+				},
+				Spec: gatewayv1beta1.HTTPRouteSpec{
+					Hostnames: []gatewayv1beta1.Hostname{"foo.com"},
+					Rules: []gatewayv1beta1.HTTPRouteRule{
+						{
+							Matches: []gatewayv1beta1.HTTPRouteMatch{
+								builder.NewHTTPRouteMatch().WithPathPrefix("/path-6").Build(),
+								builder.NewHTTPRouteMatch().WithHeaderRegex("foo", "bar[[").Build(),
+								builder.NewHTTPRouteMatch().WithPathPrefix("/path-7").Build(),
+							},
+						},
+					},
+					CommonRouteSpec: gatewayv1beta1.CommonRouteSpec{
+						ParentRefs: []gatewayv1beta1.ParentReference{{
+							Namespace: (*gatewayv1beta1.Namespace)(&managedGateway.Namespace),
+							Name:      gatewayv1beta1.ObjectName(managedGateway.Name),
+						}},
+					},
+				},
+			},
+			WantCreateErr:          true,
+			WantCreateErrSubstring: "regex parse error:\n    bar[[\n        ^\nerror: unclosed character class)",
+		},
+	)
+	testHTTPRouteValidationWebhook(ctx, t, namespace, gatewayClient, testCases)
+}
+
+func setUpEnvForTestingHTTPRouteValidationWebhook(ctx context.Context, t *testing.T) (
+	namespace string,
+	gatewayClient *gatewayclient.Clientset,
+	managedGateway *gatewayv1beta1.Gateway,
+	unmanagedGateway *gatewayv1beta1.Gateway,
+) {
+	ensureAdmissionRegistration(
+		ctx,
+		t,
+		"kong-validations-gateway",
+		[]admregv1.RuleWithOperations{
+			{
+				Rule: admregv1.Rule{
+					APIGroups:   []string{"gateway.networking.k8s.io"},
+					APIVersions: []string{"v1beta1"},
+					Resources:   []string{"httproutes"},
+				},
+				Operations: []admregv1.OperationType{admregv1.Create, admregv1.Update},
+			},
+		},
+	)
+
+	t.Log("creating a gateway client")
+	gatewayClient, err := gatewayclient.NewForConfig(env.Cluster().Config())
+	require.NoError(t, err)
+
+	ns, cleaner := helpers.Setup(ctx, t, env)
+	namespace = ns.Name
+	t.Logf("created namespace: %q", namespace)
```

Comment:
Sorry for that: it seems that https://github.com/Kong/kubernetes-ingress-controller/blob/d781042d455b82e079194728d72c4d451f9422d6/test/internal/helpers/setup.go#L41 already prints the namespace and the test name

Reply:
👍 so reverting